testsuite/classlibrary/TestSoundFile.sc

```diff
@@ -48,9 +48,93 @@ TestSoundFile : UnitTest {
 		// collect all instance variables into a Dictionary, remove problematic 'fileptr'
 		instVars = soundFile.getSlots.asDict;
 		instVars.removeAt(\fileptr);
-		
+
 		this.assertEquals(instVars, info, "SoundFile information should match that of actual sound file");
 
 	}
 
+	// tests several features:
+	// openRead and openWrite class methods (not tested above)
+	// readData and writeData
+	test_readWrite {
+		var path = PathName.tmp +/+ "SoundFile-test%.aiff".format(this.hash.asHexString(8)),
+		file, written = Signal.fill(100, { 1.0.rand2 }), read;
+
+		file = SoundFile.openWrite(path, "aiff", "float", 1, 44100);
+		if(file.isNil) {
+			this.assert(false, "SoundFile.openWrite into temp directory should not fail");
+		};
+		try {
+			file.writeData(written);
+		} { |error|
+			this.assert(false, "SoundFile:writeData should not fail with error " ++ error.errorString);
+		};
+		file.close;
+
+		file = SoundFile.openRead(path);
+		this.assertEquals(file.numFrames, written.size, "SoundFile size should match data size written by writeData");
+
+		try {
+			read = Signal.newClear(file.numFrames);
+			file.readData(read);
+		} { |error|
+			this.assert(false, "SoundFile:readData should not fail with error " ++ error.errorString);
+		};
+		file.close;
+
+		this.assertEquals(written, read, "SoundFile:readData should get the same data that were written");
+
+		File.delete(path);
+	}
+
+}
+
+TestSoundFile_Normalize : UnitTest {
+	var soundFile, path, written;
+
+	setUp {
+		path = PathName.tmp +/+ "testSFNorm%.aiff".format(this.hash.asHexString(8));
+		soundFile = SoundFile(path);
+		written = Signal.series(100, -0.5, 1/99);
+		soundFile.openWrite;
+		soundFile.writeData(written);
+		soundFile.close;
+	}
+
+	tearDown {
+		soundFile.close;
+		File.delete(path);
+	}
+
+	test_channelPeaks {
+		var peaks;
+		soundFile.openRead;
+		if(soundFile.isOpen) {
+			peaks = soundFile.channelPeaks;
+			this.assertEquals(peaks, [0.5], "Channel peaks test should read 0.5");
+		} {
+			this.assert(false, "TestSoundFile_Normalize should be able to read the test file");
```

Comment:
openRead is already tested above; why are you testing it again here?

Reply:
I'm not sure, then, what the test method should do if there's a problem reading or writing at the temp directory. The test shouldn't be successful, I'm pretty sure, but it also shouldn't report a failure in the retrieved values because it didn't retrieve any values.

On the other hand, if temp directory read/write fails here, then it will also have failed (or will fail) in TestSoundFile. So it may be reasonable simply to delete this `assert`, fail silently and allow the other test to report the problem.

I'll go ahead and remove it in a separate commit so that it can be reverted later if desired. Though it feels wrong to me somehow to have a test that might fail to complete successfully and report no failure.